app/components/embed/media_component.rb

```diff
@@ -7,5 +7,22 @@ def initialize(viewer:)
     end
 
     attr_reader :viewer
+
+    delegate :purl_object, to: :viewer
+    delegate :druid, to: :purl_object
+
+    def include_transcripts
+      Settings.enabled_features.transcripts || params[:transcripts] == 'true'
+    end
+
+    def primary_files_count
+      purl_object.contents.count do |resource|
+        resource.primary_file.present?
+      end
+    end
+
+    def many_primary_files
```

Comment:
```suggestion
    def many_primary_files?
```

Reply:
But then we can't pass it as an implicit argument `many_primary_files:`